tests/end-to-end/director/application_templates_api_test.go

```diff
@@ -0,0 +1,155 @@
+package director
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateApplicationTemplate(t *testing.T) {
+	// GIVEN
+	ctx := context.Background()
+	name := "app-template-name"
+	appTemplateInput := fixApplicationTemplate(name)
+	appTemplate, err := tc.graphqlizer.ApplicationTemplateInputToGQL(appTemplateInput)
+	require.NoError(t, err)
+
+	createApplicationTemplateRequest := fixCreateApplicationTemplateRequest(appTemplate)
+	output := graphql.ApplicationTemplate{}
+
+	// WHEN
+	t.Log("Create application template")
+
+	err = tc.RunOperation(ctx, createApplicationTemplateRequest, &output)
+	require.NoError(t, err)
+	require.NotEmpty(t, output.ID)
+	defer deleteApplicationTemplate(t, ctx, output.ID)
+
+	//THEN
+	require.NotEmpty(t, output.Name)
+	saveExample(t, createApplicationTemplateRequest.Query(), "create application template")
+
+	t.Log("Check if application template was created")
+
+	getApplicationTemplateRequest := fixApplicationTemplateRequest(output.ID)
+	appTemplateOutput := graphql.ApplicationTemplate{}
+
+	err = tc.RunOperation(ctx, getApplicationTemplateRequest, &appTemplateOutput)
+
+	require.NotEmpty(t, appTemplateOutput)
+	assertApplicationTemplate(t, appTemplateInput, appTemplateOutput)
+	saveExample(t, getApplicationTemplateRequest.Query(), "query application template")
+}
+
+func TestUpdateApplicationTemplate(t *testing.T) {
+	// GIVEN
+	ctx := context.Background()
+	name := "app-template"
+	newName := "new-app-template"
+	newDescription := "new description"
+	newAppCreateInput := &graphql.ApplicationCreateInput{
+		Name: "new-app-create-input",
+	}
+
+	t.Log("Create application template")
+	appTemplate := createApplicationTemplate(t, ctx, name)
```

Comment:
I'd defer application template deletion right after the creation because if update fails we're leaving application template in db.